### PR TITLE
Add errinfo package with error types

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/internal/errinfo/errinfo.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// +build go1.13
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package errinfo
```
Comment (jhendrixMSFT) [inline sdk/internal/errinfo/errinfo.go:6]: Open to other package names.  `errors` would be ideal but collides with the same from the standard library.

Comment (RickWinter) [reply sdk/internal/errinfo/errinfo.go:6]: I don't love `err`.  I would avoid the appreviation on the package name `errorinfo`

Comment (jhendrixMSFT) [reply sdk/internal/errinfo/errinfo.go:6]: Agreed, will fix.

**Review by RickWinter — APPROVED** [review]

**Review by richardpark-msft — APPROVED** [review]
Seems fine for `internal`.

**Pull request merged**